modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/uri/DispatcherUtil.java

```diff
@@ -118,13 +122,17 @@ public static String getServiceBasePath(ServiceInfo service) {
         return basePath;
     }
 
-    public static String concatValues(List<String> stringValues) {
+    public static String concatValues(List<String> stringValues, boolean spaceSeparated) {
         StringBuilder builder = new StringBuilder();
 
         for (int x = 0; x < stringValues.size(); ++x) {
             builder.append(stringValues.get(x));
             if (x != stringValues.size() - 1) {
-                builder.append(", ");
+                if (spaceSeparated) {
+                    builder.append(" ");
+                } else {
+                    builder.append(", ");
+                }
             }
```

Comment:
Avoid this if else check within the loop. We can initialize the separator outside the loop based on the "spaceSeparated" value and then use it for the append